Sources/Action.swift

```diff
@@ -174,6 +184,36 @@ extension Action: ActionProtocol {
 	}
 }
 
+extension ActionProtocol where Input == Void {
+	/// Initializes an action that uses an optional property for its input,
+	/// and is disabled whenever the input is nil. When executed, a SignalProducer
+	/// is created with the current value of the input.
+	///
+	/// - parameters:
+	///   - input: An Optional property whose current value is used as input
+	///            whenever the action is executed. The action is disabled
+	///            whenever the value is nil.
+	///   - execute: A closure to return a new SignalProducer based on the
+	///              current value of `input`.
+	public init<P: PropertyProtocol, T>(input: P, _ execute: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T? {
+		self.init(enabledIf: input.map { $0 != nil }) {
+			execute(input.value!)
```

Comment:
What do you think about https://github.com/ReactiveCocoa/ReactiveSwift/pull/22/commits/07ea7e7c18c04c19275e3ddcb3dc1e2bde9bc47e? It synchronises checking for `isEnabled` and calling `executeClosure`, so I think should solve this particular problem.

It feels like there could be a more fundamental refactoring of Action that would remove the need for force-unwrapping (like you were hinting at), but this is the smallest change I could think of.